File: remote/src/main/java/com/kelsos/mbrc/events/ui/ShuffleChange.java
package com.kelsos.mbrc.events.ui;

import com.kelsos.mbrc.annotations.Shuffle;

public class ShuffleChange {

  private String shuffleState;

  public ShuffleChange(@Shuffle.State String shuffleState) {
    this.shuffleState = shuffleState;
  }

  @Shuffle.State public String getShuffleState() {
    return this.shuffleState;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/events/ui/ShuffleChange.kt
package com.kelsos.mbrc.events.ui

import android.support.annotation.StringDef

class ShuffleChange(@ShuffleState val shuffleState: String) {

  @StringDef(OFF, AUTODJ, SHUFFLE)
  @Retention(AnnotationRetention.SOURCE)
  annotation class ShuffleState

  companion object {
    const val OFF = "off"
    const val AUTODJ = "autodj"
    const val SHUFFLE = "shuffle"
  }
}
